test/print_test_stats.py

```diff
@@ -156,26 +183,67 @@ def send_report_to_s3(reports, *, total_seconds):
     # input files of a few megabytes in size like these JSON files, and
     # because for some reason zlib doesn't seem to play nice with the
     # gunzip command whereas Python's bz2 does work with bzip2
-    obj.put(Body=bz2.compress(json.dumps({
-        **build_info(),
-        'total_seconds': total_seconds,
-        'suites': {
-            name: {
-                'total_seconds': suite.total_time,
-                'cases': [
-                    {
-                        'name': case.name,
-                        'seconds': case.time,
-                        'errored': case.errored,
-                        'failed': case.failed,
-                        'skipped': case.skipped,
-                    }
-                    for case in suite.test_cases
-                ],
-            }
-            for name, suite in reports.items()
-        }
-    }).encode()))
+    obj.put(Body=bz2.compress(json.dumps(obj).encode()))
+
+def print_regressions(obj, *, n, stdev_threshold):
+    sha1 = os.environ.get("CIRCLE_SHA1")
+
+    base = subprocess.check_output(
+        ["git", "merge-base", sha1, "origin/master"],
+        encoding="ascii",
+    ).strip()
+
+    count_spec = f"{base}..{sha1}"
+    intermediate_commits = int(subprocess.check_output(
+        ["git", "rev-list", "--count", count_spec],
+        encoding="ascii"
+    ))
+    ancestry_path = int(subprocess.check_output(
+        ["git", "rev-list", "--ancestry-path", "--count", count_spec],
+        encoding="ascii",
+    ))
+
+    # if current commit is already on master, we need to exclude it from
+    # this history; otherwise we include the merge-base
+    commits = subprocess.check_output(
+        ["git", "rev-list", f"--max-count={n+1}", base],
+        encoding="ascii",
+    ).splitlines()
+    on_master = False
+    if base == sha1:
+        on_master = True
+        commits = commits[1:]
+    else:
+        commits = commits[:-1]
+
+    job = os.environ.get("CIRCLE_JOB")
+    s3 = boto3.resource("s3")
+    bucket = s3.Bucket(name="ossci-metrics")
+    index = {}
+    for commit in commits:
+        summaries = bucket.objects.filter(Prefix=f"test_time/{commit}/{job}/")
+        index[commit] = list(summaries)
+
+    objects = {}
+    # should we do these in parallel?
+    for commit, summaries in index.items():
+        objects[commit] = []
+        for summary in summaries:
+            binary = summary.get()["Body"].read()
+            string = bz2.decompress(binary).decode("utf-8")
+            objects[commit].append(json.loads(string))
+
+    print()
```

Comment:
is this just for a line separator?

Reply:
yep

Reply:
Ah. Cool. Could you add msg like `----- Historic stats comparison result ------`

Reply:
There is already a message (shown in the pastebin); it looks like this:

```
Following output is to check for test time regressions:
    job: <job>
    commit: <sha1>
```

Do you want me to change it (e.g. by adding more hyphens)?

Reply:
no it is fine